firestore/src/common/document_reference.cc

```diff
@@ -113,12 +114,23 @@ CollectionReference DocumentReference::Parent() const {
 
 CollectionReference DocumentReference::Collection(
     const char* collection_path) const {
+  if (!collection_path) {
+    SimpleThrowInvalidArgument("Collection path cannot be null.");
+  }
+  if (collection_path[0] == '\0') {
```

Comment:
Is there a reason to not use `strlen(collection_path) == 0` here, instead of `collection_path[0] == '\0'`?

Reply:
I don't have a strong preference on this -- it just seemed a little simpler to do.

Reply:
Personally, I'd prefer `strlen` because IMO it is more readable; that is, it directly conveys what you are checking. Checking that the first element == 0 is indirectly checking the string's length. This won't block my approval though.